tests/e2e/upgrades_test.go

```diff
@@ -0,0 +1,396 @@
+/*
+Copyright © 2022 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/types"
+	kubectl "github.com/rancher-sandbox/ele-testhelpers/kubectl"
+	upgradev1 "github.com/rancher/system-upgrade-controller/pkg/apis/upgrade.cattle.io/v1"
+
+	"github.com/rancher-sandbox/rancheros-operator/tests/catalog"
+)
+
+const cattleNamespace = "cattle-system"
+const fleetNamespace = "fleet-local"
+const discoveryPluginImage = "quay.io/costoolkit/upgradechannel-discovery:v0.3-4b83dbe"
+
+func getPlan(s string) (up *upgradev1.Plan, err error) {
+	up = &upgradev1.Plan{}
```

Comment:
I've created https://github.com/rancher-sandbox/ele-testhelpers/pull/3 to refactor those common funcs, once merged I'll consume them here in place of the dups here

Reply:
done